File: AspNetMvcUnleashed/0672329980_source/Chapter11/VB/MvcApplication1/MvcApplication1/Controllers/SimpleMovieController.vb
Public Class SimpleMovieController
    Inherits Controller

    Private _service As ISimpleMovieService

    Public Sub New()
        Me.New(New SimpleMovieService())
    End Sub

    Public Sub New(ByVal service As ISimpleMovieService)
        _service = service
    End Sub

    Public Function Index() As ActionResult
        Return View(_service.ListMoviesCached())
    End Function

End Class

File: AspNetMvcUnleashed/0672329980_source/Chapter11/CS/MvcApplication1/MvcApplication1/Controllers/SimpleMovieController.cs
using System.Web.Mvc;
using MvcApplication1.Models;

namespace MvcApplication1.Controllers
{
    public class SimpleMovieController : Controller
    {
        private ISimpleMovieService _service;

        public SimpleMovieController() 
            : this(new SimpleMovieService()) { }

        public SimpleMovieController(ISimpleMovieService service)
        {
            _service = service;
        }

        public ActionResult Index()
        {
            return View(_service.ListMoviesCached());
        }
    }
}
